store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreNamedOperationTest.java

```diff
@@ -0,0 +1,210 @@
+import uk.gov.gchq.gaffer.named.operation.AddNamedOperation;
+import uk.gov.gchq.gaffer.named.operation.NamedOperation;
+import uk.gov.gchq.gaffer.operation.OperationChain;
+import uk.gov.gchq.gaffer.operation.OperationException;
+import uk.gov.gchq.gaffer.operation.impl.get.GetAllElements;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.GRAPH_ID_TEST_FEDERATED_STORE;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.contextTestUser;
+import static uk.gov.gchq.gaffer.federatedstore.FederatedStoreTestUtil.resetForFederatedTests;
+
+public class FederatedStoreNamedOperationTest {
+
+    public static final String SELF_REFERENCING_NAMED_OP = "selfRefNamedOp";
+    public static final String OUTER = "outer";
+    public static final String INNER = "inner";
+
+    @BeforeEach
+    public void before() {
+        resetForFederatedTests();
+    }
+
+    @Test
+    public void shouldFailWhenAddingSelfReferencingNamedOperationWhenNestedNamedOpsIsAllowed() throws Exception {
+
+        final FederatedStoreProperties properties = new FederatedStoreProperties();
+        properties.setNestedNamedOperationAllow(true);
+
+        final Graph federatedStore = new Graph.Builder()
+                .config(new GraphConfig(GRAPH_ID_TEST_FEDERATED_STORE))
+                .addSchema(new Schema())
+                .storeProperties(properties)
+                .build();
+
+        assertThatExceptionOfType(OperationException.class)
+                .isThrownBy(() ->
+                        federatedStore.execute(new AddNamedOperation.Builder()
+                                .name(SELF_REFERENCING_NAMED_OP)
+                                .operationChain(new OperationChain.Builder()
+                                        .first(new NamedOperation.Builder()
+                                                .name(SELF_REFERENCING_NAMED_OP)
+                                                .build())
+                                        .build())
+                                .build(), contextTestUser()))
+                .withMessageContaining("Self referencing namedOperations would cause infinitive loop. operationName:selfRefNamedOp");
+
+        assertThatExceptionOfType(UnsupportedOperationException.class)
+                .isThrownBy(() -> federatedStore.execute(new NamedOperation.Builder()
+                        .name(SELF_REFERENCING_NAMED_OP)
+                        .build(), contextTestUser()))
+                .withMessageContaining("The named operation: selfRefNamedOp was not found.");
+    }
+
+    @Test
+    public void shouldFailWhenAddingAnyNestedNamedOperationWhenNestedNamedOpsIsNotAllowed() throws Exception {
+
+        final FederatedStoreProperties properties = new FederatedStoreProperties();
+        properties.setNestedNamedOperationAllow(false);
```

Comment:
Could we use `setNestedNamedOperationAllow` here instead, as that method doesn't have test coverage.